docs/README.md

```diff
@@ -695,9 +696,42 @@ With capturing, the event is first captured by the outermost element and propaga
 https://www.youtube.com/watch?v=-bS6u_oQFtc&list=PL7pEw9n3GkoW5bYOhVAtmJlak3ZK7SaDf&index=29
 
 
+## 11. Generator function
+
+Function keyword followed by an asterisk is used to define a generator function, which returns a Generator object.
+We can exit and re-entered the generator function later on. In case of re-entrances, their context (variable bindings) will be saved.
+Important point to note here is that calling a generator function does not execute its body immediately, in fact it returns an iterator object for the function.
+In short, a generator appears to be a function but it behaves like an iterator.
+
+```
+Syntax :
+
+function* generatorFunctionName([param[, param[, ... param]]]) {
+   statements
+}
+```
+```
+name : Function name
+param | Optional : Formal parameter for the function.
+statements : Comprising the body of the function.
+```
+```
+  function* generator(i) {
+  yield i;
+  yield i + 1;
+}
+const gen = generator(1);
+console.log(gen.next().value);
+// expected output: 1
+console.log(gen.next().value);
+// expected output: 2 </div>
+```
+* Generators are a special class of functions that simplify the task of writing iterators.
+* A generator is a function that produces a sequence of results instead of a single value, i.e you generate ​a series of values.
```

Comment:
Could you mention these 2 points in the beginning

Reply:
sure